src/main/java/org/tikv/common/log/SlowLogImpl.java

```diff
@@ -68,45 +64,71 @@ public synchronized SlowLogSpan start(String name) {
     return slowLogSpan;
   }
 
+  @Override
+  public long getTraceId() {
+    return traceId;
+  }
+
+  @Override
+  public long getThresholdMS() {
+    return slowThresholdMS;
+  }
+
   @Override
   public void setError(Throwable err) {
     this.error = err;
   }
 
   @Override
   public void log() {
+    if (error != null || timeExceeded()) {
+      logger.warn("SlowLog:" + getSlowLogJson());
+    }
+  }
+
+  boolean timeExceeded() {
     long currentNS = System.nanoTime();
-    long currentMS = startMS + (currentNS - startNS) / 1_000_000;
-    if (error != null || (slowThresholdMS >= 0 && currentMS - startMS > slowThresholdMS)) {
-      logger.warn("SlowLog:" + getSlowLogString(currentMS));
+    long durationMS = (currentNS - startNS) / 1_000_000;
+    if (slowThresholdMS >= 0 && durationMS > slowThresholdMS) {
+      SimpleDateFormat dateFormat = new SimpleDateFormat("HH:mm:ss.SSS");
+      logger.warn(
```

Comment:
please remove this log, only print one log when `log()` is called.

Reply:
Fixed. The tow logs are merged.